File: client/src/app/modules/miscellaneous/services/department.vat.setting.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { DepartmentVatSetting } from '../../miscellaneous/model/DepartmentVatSetting';

@Injectable({
  providedIn: 'root'
})
export class DepartmentVatSettingService extends AppRestService<DepartmentVatSetting>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getDepartmentVatSettings() {
    return this.getList(new HttpParams(), "department-vat-settings");
  }
  
   addOrUpdateDepartmentVatSetting(departmentVatSetting: DepartmentVatSetting) {
    return this.addOrUpdate(new HttpParams(), "department-vat-setting", departmentVatSetting, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/DepartmentVatSettingService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.Department;
import com.jss.osiris.modules.osiris.miscellaneous.model.DepartmentVatSetting;

public interface DepartmentVatSettingService {
    public List<DepartmentVatSetting> getDepartmentVatSettings();

    public DepartmentVatSetting getDepartmentVatSetting(Integer id);

    public DepartmentVatSetting addOrUpdateDepartmentVatSetting(DepartmentVatSetting departmentVatSetting);

    public DepartmentVatSetting getDepartmentVatSettingByDepartment(Department department);
}
